engine/items/inventory.asm

```diff
@@ -0,0 +1,658 @@
+INCLUDE "constants.asm"
+
+SECTION "AddItemToInventory_", ROMX[$4AA1], BANK[$03]
+
+_ReceiveItem: ; 03:4AA1
+	call DoesHLEqualwNumBagItems
+	jp nz, PutItemInPocket ; If its not bag, then its the PC, so jump down
+	push hl
+	ld hl, CheckItemPocket
+	ld a, BANK( CheckItemPocket )
+	call FarCall_hl
+	ld a, [wItemAttributeParamBuffer] ; a = PocketType
+	dec a
+	ld hl, .Pockets
+	jp CallJumptable
+	
+.Pockets: ; 03:4ABA
+	dw .Item
+	dw .KeyItem
+	dw .Ball
+	dw .TMHM
+
+.Item: ; 03:4AC2
+	pop hl
+	jp PutItemInPocket
+
+.KeyItem: ; 03:4AC6
+	pop hl
+	jp ReceiveKeyItem
+
+.Ball: ; 03:4ACA
+	pop hl
+	ld a, [wCurItem]
+	ld c, a
+	call GetBallIndex ; get the index of the ball being added
+	jp ReceiveBall ; increase the quantity at that index
+
+.TMHM: ; 03:4AD5
+	pop hl
+	ld a, [wCurItem]
+	ld c, a
+	call GetTMHMNumber ; get the index of the TMHM being added
+	jp ReceiveTMHM ; increase the quantity at that index
+	
+_TossItem: ; 03:4AE0
+	call DoesHLEqualwNumBagItems
+	jr nz, .removeItem ; If its not bag, then its the PC, so jump down
+	push hl
+	ld hl, CheckItemPocket
+	ld a, BANK( CheckItemPocket )
+	call FarCall_hl
+	ld a, [wItemAttributeParamBuffer] ; a = PocketType
+	dec a
+	ld hl, .Pockets
+	jp CallJumptable
+	
+.Pockets ; 03:4AF8
+	dw .Item
+	dw .KeyItem
+	dw .Ball
+	dw .TMHM
+	
+.Ball ; 03:4B00
+	pop hl
+	ld a, [wCurItem]
+	ld c, a
+	call GetBallIndex ; get the index of the ball being removed
+	jp TossBall ; and decrease the quantity at that index
+	
+.TMHM ; 03:4B0B
+	pop hl
+	ld a, [wCurItem]
+	ld c, a
+	call GetTMHMNumber ; get the index of the TMHM being removed
+	jp TossTMHM ; and decrease the quantity at that index
+	
+.KeyItem ; 03:4B16
+	pop hl
+	jp TossKeyItem
+	
+.Item ; 03:4B1A
+	pop hl
+
+.removeItem ; 03:4B1B
+	jp RemoveItemFromPocket
+
+_CheckItem:: ; 03:4B1E
+	call DoesHLEqualwNumBagItems
+	jr nz, .checkItem ; If its not bag, then its the PC, so jump down
+	push hl
+	ld hl, CheckItemPocket
+	ld a, BANK( CheckItemPocket )
+	call FarCall_hl
+	ld a, [wItemAttributeParamBuffer] ; a = PocketType
+	dec a
+	ld hl, .Pockets
+	jp CallJumptable
+	
+.Pockets ; 03:4B36
+	dw .Item
+	dw .KeyItem
+	dw .Ball
+	dw .TMHM
+	
+.Ball ; 03:4B3E
+	pop hl
+	ld a, [wCurItem]
+	ld c, a
+	call GetBallIndex ; get the index of the ball being checked
+	jp CheckBall ; and check that index
+
+.TMHM ; 03:4B49
+	pop hl
+	ld a, [wCurItem]
+	ld c, a
+	call GetTMHMNumber ; get the index of the TMHM being checked
+	jp CheckTMHM ; and check that index
+	
+.KeyItem ; 03:4B54
+	pop hl
+	jp CheckKeyItems
+	
+.Item ; 03:4B58
+	pop hl
+	
+.checkItem
+	jp CheckTheItem
+
+; Zero flag is set if hl = wNumBagItems
+DoesHLEqualwNumBagItems: ; 03:4B5C
+	ld a, l
+	cp LOW(wNumBagItems)
+	ret nz
+	ld a, h
+	cp HIGH(wNumBagItems)
+	ret
+
+PutItemInPocket: ; 03:4B64
+	ld d, h
+	ld e, l ; de = address of start of inventory
+	inc hl
+	ld a, [wCurItem]
+	ld c, a
+	ld b, 0 ; b = available space for slots already containing this item, initialize to 0
+	
+.findItemLoop

```

Comment:
We don't camelCase around these parts. Use snake_case for local labels. I'm not going to mention every instance of this. The only exception here are the pockets, which have appropriate PascalCase names.